src/app/features/students/components/student-modal/student-modal.component.ts

```diff
@@ -0,0 +1,114 @@
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output, Signal, computed, effect, inject } from '@angular/core';
+import { FormGroup, FormsModule, NonNullableFormBuilder, ReactiveFormsModule, Validators } from '@angular/forms';
+import { CommonModule } from '@angular/common';
+
+import { NzFormModule } from 'ng-zorro-antd/form';
+import { NzInputModule } from 'ng-zorro-antd/input';
+import { NzModalModule } from 'ng-zorro-antd/modal';
+
+import { Student, StudentFormValue, StudentModalType } from '../../../../shared/interfaces/students';
+import { StudentService } from '../../../../shared/services/student.service';
+import { NzMessageModule, NzMessageService } from 'ng-zorro-antd/message';
+import { Subject, finalize, takeUntil, tap } from 'rxjs';
+
+
+@Component({
+  selector: 'app-student-modal',
+  standalone: true,
+  imports: [NzModalModule, NzFormModule, NzInputModule, ReactiveFormsModule, FormsModule, CommonModule, NzMessageModule],
+  templateUrl: './student-modal.component.html',
+  styleUrl: './student-modal.component.sass'
+})
+export class StudentModalComponent implements OnInit, OnDestroy {
+  @Input() public isVisible: boolean = false;
+  @Input() public action: StudentModalType = "add";
+  @Output() public visibleChanged = new EventEmitter<boolean>();
+
+  private fb: NonNullableFormBuilder = inject(NonNullableFormBuilder)
+  private studentService: StudentService = inject(StudentService);
+  private toast: NzMessageService = inject(NzMessageService)
+  private student: Signal<Student | null> = computed(() => this.studentService.selectedStudent());
+
+  private onDestroy$: Subject<void> = new Subject();
+
+  private error: Signal<string | null> = computed(() => this.studentService.error())
+
+
+  public isLoading: boolean = false;
+
+  public validateForm: FormGroup<StudentFormValue> = this.fb.group({
+    firstName: ['', [Validators.minLength(1), Validators.required]],
+    lastName: ['', [Validators.minLength(1), Validators.required]],
+    group: [0, [Validators.min(0), Validators.required]]
+  });
+
+
+  constructor() {
+    effect(() => {
+      if (this.student()) {
+        this.validateForm.patchValue({ ...this.student() })
```

Comment:
Let's move form initialization in ngOnInit

Reply:
effect will work only in constructor